### PR TITLE
Repo brancher: Retry git commands

### PR DESCRIPTION
**Pull request by alvaroaleman**
This PR factors out the git command execution into a func that does
retrying. It also moves all the command-related logging into that func.

ref https://issues.redhat.com/browse/DPTP-1625

/assign @droslean

### CODE REVIEW

File: cmd/repo-brancher/main.go [hunk 1]
```diff
@@ -235,3 +204,33 @@ func main() {
...
+			}
+			break
+		}
+		l = l.WithField("output", string(b))
+		if err != nil {
+			l.Error("Failed to execute command.")
+		} else {
+			l.Debug("Executed command.")
+		}
+		return err
```
Comment (droslean) [inline cmd/repo-brancher/main.go:234]: Can we just do this instead ?
```suggestion
		if err != nil {
			l.Error("Failed to execute command.")
			return err
		} 
		l.Debug("Executed command.")
		return nil
```

Comment (alvaroaleman) [reply cmd/repo-brancher/main.go]: good point, updated

**Review by droslean — APPROVED** [review]

File: cmd/repo-brancher/main.go [hunk 2]
```diff
@@ -151,66 +147,39 @@ func main() {
...
-			cmd.Dir = repoDir
-			cmdLogger.Debug("Running command.")
-			if out, err := cmd.CombinedOutput(); err != nil {
-				cmdLogger.WithError(err).WithFields(logrus.Fields{"output": string(out)}).Error("Failed to execute command.")
+			if err := executeGitCmd(logger, command...); err != nil {
 				failed = true
 				continue
-			} else {
-				cmdLogger.WithFields(logrus.Fields{"output": string(out)}).Debug("Executed command.")
-				if string(out) == "" && !o.fastForward {
```
Comment (wking) [inline cmd/repo-brancher/main.go:166]: This drops the only `fastForward` consumer that I can find.  Should we deprecate `--fast-forward`, drop it from openshift/release, and then drop the command-line option here?

**Pull request merged**